### PR TITLE
Goow 4

### PR DESCRIPTION
**Pull request by beserkzombie**

### CODE REVIEW

File: TileMap.gd [hunk 1]
```diff
@@ -528,3 +536,29 @@ func get_required_and_free_doors_to_palce(doorsToAdd, forcedDoorsToPlace, curren
 		set_tile(currentDirectionMidPoint.x, currentDirectionMidPoint.y, directionWallNeeded[workingDirection])
 	
 	return [doorsToAdd, forcedDoorsToPlace]
+
+func place_treasure_chest(anchorPosition):
+	#generatePossible chest spawnPoints
+	var wallAndDoorOffset = 2
+	#remove spots where wall are	
+	#remove spots where a door could be
```
Comment (demwilson) [inline TileMap.gd:544]: Does this account for possibly placing it on the stairs/ladder?

File: TileMap.gd [hunk 2]
```diff
@@ -528,3 +536,29 @@ func get_required_and_free_doors_to_palce(doorsToAdd, forcedDoorsToPlace, curren
...
+	var possibleSpawnPoints = []
+	for possibleSpawnPointX in range(roomCellSize.x - wallAndDoorOffset * 2):
+		for possibleSpawnPointY in range(roomCellSize.y - wallAndDoorOffset * 2):
+			possibleSpawnPoints.append(anchorPosition + Vector2(possibleSpawnPointX + wallAndDoorOffset, possibleSpawnPointY + wallAndDoorOffset))
+	#randomly pic a position where the chest should be
+	var spawnPoint = possibleSpawnPoints[randi() % possibleSpawnPoints.size()]
+	#set the tile at the position to closed chest
+	set_tile(spawnPoint.x, spawnPoint.y, Tile.CLOSED_CHEST)
+	pass
```
Comment (demwilson) [inline TileMap.gd]: Should delete this `pass` line.

File: TileMap.gd [hunk 3]
```diff
@@ -528,3 +536,29 @@ func get_required_and_free_doors_to_palce(doorsToAdd, forcedDoorsToPlace, curren
...
+func open_treasure_chest(chestPosition):
+	#prevent player
+	overworld.player.set_can_move(false)
+	#prevent collision
+	chestIsOpen = true
+	#set tile at the position to open chest
+	set_tile(chestPosition.x, chestPosition.y, Tile.OPEN_CHEST)	
+	#get loot
+	overworld.get_loot_for_chest(levelNum)
+	pass
```
Comment (demwilson) [inline TileMap.gd]: Should delete this `pass` line as well.

**Review by demwilson — CHANGES_REQUESTED** [review]

File: TileMap.gd (hunk 1 above)
Comment (beserkzombie) [reply TileMap.gd:544]: yes, because stair/ladder only spawns in a possible starting spot

**Pull request merged**